app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -115,14 +115,50 @@ def leave_membership
     end
   end
 
+  def update_membership_role
+    @chat_channel = ChatChannel.find_by(id: params[:id])
+    authorize @chat_channel, :update?
+    membership = ChatChannelMembership.find_by(id: channel_membership_params[:membership_id], chat_channel_id: @chat_channel.id)
+
+    result = UpdateChatChannelMembershipService.perform(@chat_channel, membership, { role: channel_membership_params[:role] })
+    if result.errors.any?
+      render json: { success: false, message: "Failed to update membership", errors: chat_channel_membership.errors.full_messages }, status: :bad_request
+    else
+      render json: { success: true, message: "User Membership is updated" }, status: :ok
+    end
+  end
+
+  def join_channel_invitation
+    chat_channel = ChatChannel.find_by(slug: params[:channel_slug])
+    authorize chat_channel
+    membership = ChatChannelMembership.find_by(user_id: current_user.id, chat_channel_id: chat_channel.id)
+
+    unless membership
+      result = CreateChatChannelMembershipService.perform(chat_channel, current_user)
+      send_chat_action_message("@#{result.user.username} join the channel", current_user, chat_channel.id, "joined") unless result&.errors&.any?
+    end
+
+    if membership && membership.status != "active"
```

Comment:
I got a little lost with this `if/unless` set of statements. I think it might make it a little more clear to have a single statement like
```ruby
if !membership 
  # Create membership
elsif membership.status != "active"
  # Update membership
end
```